### PR TITLE
libusb: backport bugfix for 1.0.24

### PR DESCRIPTION
**Pull request by yorickvP**
Both arch and debian apply this same patch. Resolves the issue at https://github.com/libusb/libusb/issues/831 .

###### Motivation for this change

I spent half the day debugging an issue no other distros were having. It turns out to be a regression that the other distros are patching. See https://bugs.archlinux.org/task/69121

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yorickvP** [comment]
@ofborg build usbutils

**Comment by alyssais** [comment]
Could we fetchpatch from one of those other distros, rather than including our own copy of the patch?

**Comment by jtojnar** [comment]
Or even from upstream: `https://github.com/libusb/libusb/commit/f6d2cb561402c3b6d3627c0eb89e009b503d9067.patch`, excluding the version file.

**Comment by yorickvP** [comment]
@alyssais possible, it would be https://raw.githubusercontent.com/archlinux/svntogit-packages/449157b747c936322f7f31ed03bada6d35259228/trunk/libusb-1.0.24-descriptor_parsing.patch , which doesn't seem much nicer to do than this.
@jtojnar: yes, that was what I ran into with upstream ;)

**Comment by alyssais** [comment]
legit, this is such a small patch that it’s not a big deal

**Review by alyssais — APPROVED** [review]

**Comment by jtojnar** [comment]
@yorickvP `fetchpatch` has `excludes` argument precisely for that.

**Comment by yorickvP** [comment]
Oh! I'll do that.

**Comment by yorickvP** [comment]
:heavy_check_mark: good to go if ofborg agrees

**Comment by yorickvP** [comment]
@ofborg build libusb

**Comment by yorickvP** [comment]
Okay, it works :)

File: pkgs/development/libraries/libusb1/default.nix [hunk 1]
```diff
@@ -22,6 +23,13 @@ stdenv.mkDerivation rec {
 
   outputs = [ "out" "dev" ];
 
+  patches = [ (fetchpatch {
+    # https://bugs.archlinux.org/task/69121
+    url = https://github.com/libusb/libusb/commit/f6d2cb561402c3b6d3627c0eb89e009b503d9067.patch;
```
Comment (prusnak) [inline pkgs/development/libraries/libusb1/default.nix:28]: URLs should be quoted: 
```suggestion
    url = "https://github.com/libusb/libusb/commit/f6d2cb561402c3b6d3627c0eb89e009b503d9067.patch";
```

**Review by prusnak — CHANGES_REQUESTED** [review]

**Review by prusnak — APPROVED** [review]

**Pull request merged**